contracts/strategies/traderjoe/TraderJoe.sol

```diff
@@ -196,15 +312,18 @@ contract TraderJoe is UUPSUpgradeable, StrategyOwnablePausableBaseUpgradeable {
         TraderJoeStorage storage strategyStorage = TraderJoeStorageLib
             .getStorage();
 
-        strategyStorage.masterChef.deposit(strategyStorage.farmId, 0);
+        strategyStorage.lbPair.collectFees(
+            address(this),
+            strategyStorage.binIds
+        );
 
         address[] memory path = new address[](2);
-        path[0] = address(strategyStorage.joeToken);
+        path[0] = address(strategyStorage.pairDepositToken);
         path[1] = address(depositToken);
 
         SwapServiceLib.swapExactTokensForTokens(
             swapService,
-            strategyStorage.joeToken.balanceOf(address(this)),
```

Comment:
can we only swap the increment here instead of the full balance?

there reason is this: let's say during addLiquidity our code makes a mistake and there is indeed a lot of pairDepositToken left on the contract balance. If in the reapReward we swap the entire balance, then that error is 'masked' (goes undetected)

Reply:
we do have to pay a little bit more gas though, maybe that is why you decided to swap the entire balance

Reply:
As you wish, I'll add commit of it